src/crush/CrushWrapper.cc

```diff
@@ -1703,25 +1703,26 @@ int CrushWrapper::add_bucket(
 				      weights);
   assert(b);
   int r = crush_add_bucket(crush, bucketno, b, idout);
+  int pos = -1 - *idout;
   for (auto& p : choose_args) {
     crush_choose_arg_map& cmap = p.second;
     if (cmap.args) {
-      if ((int)cmap.size <= *idout) {
+      if ((int)cmap.size <= pos) {
 	cmap.args = (crush_choose_arg*)realloc(
 	  cmap.args,
-	  sizeof(crush_choose_arg) * (*idout + 1));
+	  sizeof(crush_choose_arg) * (pos + 1));
```

Comment:
the `if (cmap.args)` checking sitting a few lines above should have guaranteed this